lesson8-UnitTesting/RecentlyUsedList/src/RecentlyUsedList.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace RecentlyUsedList
+{
+    public class RecentlyUsedListSolver
+    {
+        public RecentlyUsedListSolver(int capacity = 5)
+        {
+            if(capacity <= 0)
+                throw new ArgumentOutOfRangeException($"Capacity should be greater than zero: {capacity}");
+            Capacity = capacity;
+        }
+        
+        private readonly List<string> _storage = new List<string>();
+        public int Count => _storage.Count;
+        public int Capacity { get;  }
+
+        public void Add(string element)
+        {
+            if(string.IsNullOrEmpty(element))
+                throw new ArgumentNullException();
+            
+            if (_storage.Contains(element))
+            {
+                _storage.Remove(element);
+            }
+
+            if (Count == Capacity)
+            {
+                _storage.Remove(_storage[0]);
+            }
+            _storage.Add(element);
+        }
+        
+        public string this[int index]
+        {
+            get
+            {
+                if(index < 0)
+                    throw new IndexOutOfRangeException();
+                return _storage[Count - 1 - index];
+            }
+            set
+            {
+                if(index < 0)
+                    throw new IndexOutOfRangeException();
+
+                _storage[Count - 1 - index] = value;
```

Comment:
I think it worth to check the value for null.

Reply:
for the index or storage?

Reply:
The one we want to set.

Reply:
done